packages/asset-swapper/src/utils/market_operation_utils/constants.ts

```diff
@@ -1303,7 +1306,14 @@ export const LIDO_INFO_BY_CHAIN = valueByChainId<LidoInfo>(
 export const BALANCER_SUBGRAPH_URL = 'https://api.thegraph.com/subgraphs/name/balancer-labs/balancer';
 export const BALANCER_TOP_POOLS_FETCHED = 250;
 export const BALANCER_MAX_POOLS_FETCHED = 3;
-export const BALANCER_V2_SUBGRAPH_URL = 'https://api.thegraph.com/subgraphs/name/balancer-labs/balancer-v2';
+
+export const BALANCER_V2_SUBGRAPH_URL_BY_CHAIN = valueByChainId<string>(
+    {
+        [ChainId.Mainnet]: 'https://api.thegraph.com/subgraphs/name/balancer-labs/balancer-v2',
+        [ChainId.Polygon]: 'https://api.thegraph.com/subgraphs/name/balancer-labs/balancer-polygon-v2',
+    },
+    NULL_ADDRESS,
```

Comment:
I pretty sure `NULL_ADDRESS` is `0x000....` here, I don't think we can use that as a fallback when it's expecting an URL. This will probably cause issues on chains that don't have Balancer v2.

We could default to `null` but then you'd need to handle the cases where there shouldn't be any cache.

Reply:
agree with kim, might spew errors on BSC

Reply:
made `Mainnet` URL by default for now